x-pack/plugins/gis/public/components/layer_panel/join_editor/resources/join_expression.js

```diff
@@ -51,10 +52,98 @@ export class JoinExpression extends Component {
     }
   }
 
+  _onLeftFieldChange = (selectedFields) => {
+    const selectedField = selectedFields.length > 0 ? selectedFields[0].value : null;
+    this.props.onLeftFieldChange(selectedField.name);
+  };
+
+  _renderLeftFieldSelect() {
+    const {
+      leftValue,
+      leftFields,
+    } = this.props;
+
+    if (!leftFields) {
```

Comment:
You could add `|| !leftFields.length` since this could be an empty array but I don't believe it breaks anything if you don't

Reply:
In this case, you would want to show the left field select that is empty so the user can see that there  are not options to choose instead of wondering why there is nothing there.